corehq/apps/user_importer/tests/test_importer.py

```diff
@@ -1425,6 +1425,15 @@ def test_upload_with_no_group(self):
         self.assertEqual(user_history.change_messages['groups'],
             UserChangeMessage.groups_info([])['groups'])
 
+    def test_create_or_update_commcare_users_and_groups_with_bad_username(self):
+        result = create_or_update_commcare_users_and_groups(
+            self.domain.name,
+            [self._get_spec(username="..bad username")],
+            self.uploading_user,
+            self.upload_record.pk,
+        )
+        self.assertEqual(result["rows"][0]["flag"], "Enter a valid email address.")
```

Comment:
is enter a valid email address the correct error in this case? the usernames for commcare users should not be email addresses

Reply:
+1 on that.

Reply:
So the reason for this is because in line 492 of importer.py, the `normalize_username` method runs an email validator against the username. This validator returns a standard message of "Enter a valid email address", and there is currently no way to catch that validation error and handle it better other than rewriting the normalization method, which is what I'm going to do after my sync with Graham today.

Reply:
got it 👍 
syncing offline: this method uses the built in `EmailValidator` to throw the error message. When this method is used, the `ValidationError` is caught and the relevant error message is thrown instead.